### PR TITLE
148 hidden things

### LINKED ISSUE #148: Add "hidden" flag to objects
As a "delete-lite" option

### PR DESCRIPTION
**Pull request by dethe**
Support for marking things "hidden" to remove them from search results and lists. You can still pull them up by id for view or edit if needed.

Fixes #148 

Also cleaned out some files that were no longer needed and other misc. cleanup.

Requires migration 29

### CODE REVIEW

File: api/sql/thing_by_id.sql [hunk 1]
```diff
@@ -15,6 +15,8 @@ WITH authors_list AS (
 full_thing AS (
   SELECT
     ${table:name}.*,
+    COALESCE(location, '("","","","","","","","","")'::geolocation) AS location,
```
Comment (davidascher) [inline api/sql/thing_by_id.sql]: this seems unrelated to hiding - what is it for?

File: api/sql/thing_by_id.sql [hunk 2]
```diff
@@ -15,6 +15,8 @@ WITH authors_list AS (
...
+    COALESCE(lead_image, '("","",0)'::attachment) AS lead_image,
```
Comment (davidascher) [inline api/sql/thing_by_id.sql]: Same here

File: migrations/migration_029.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
+-- Fix instances of "null" vs. null
```
Comment (davidascher) [inline migrations/migration_029.sql]: this is duplicating a bunch of `_028`, which doesn't seem right.

File: api/sql/thing_by_id.sql (hunk 1 above)
Comment (dethe) [inline api/sql/thing_by_id.sql]: Ensures that we always return an empty location rather than null. We do this everywhere else, but I'd missed this one, added for consistency.

File: api/sql/thing_by_id.sql (hunk 2 above)
Comment (dethe) [inline api/sql/thing_by_id.sql]: OK, had missed more than one.

File: migrations/migration_029.sql (hunk 3 above)
Comment (dethe) [inline migrations/migration_029.sql]: Changes in _28 hadn't caught everything I thought they would, so this fixes the fix.
